src/Services/DataService/DataService.ts

```diff
@@ -176,31 +176,43 @@ export abstract class DataService<T extends IModelData, R = T> extends BaseServi
     const itemObservables = ids.map(id => this.getById(id));
 
     const observable = combineLatest$(...itemObservables)
-      .auditTime(25)
-      .startWith([]);
+      .auditTime(25);
 
     this.observablesByIdsCache[cacheKey] = observable;
     return observable;
   }
 
-  public getByQuery(queryParams): Observable<IModel<T>[]> {
-    const hashParams = hash(queryParams || {});
+  public getByQuery(queryBuilder: IQueryBuilder): Observable<IQueryManager<T>> {
+    const hashParams = queryBuilder.getHashCode();
+
     if (hashParams in this.observablesByQueryCache) {
       return this.observablesByQueryCache[hashParams];
     }
 
-    this.actions.fetchAll(queryParams).invoke();
-
     const observable = BaseService
       .getStateObservable()
-      .map((state) => this.selectors.getItems(state, queryParams))
+      .map((state) => state[this.name].requestCache.get(hashParams))
       .distinctUntilChanged()
-      .map(items => (items != null && "toJS" in items) ? items.toJS() : items)
       .shareReplay(1);
 
-    this.observablesByQueryCache[hashParams] = observable;
+    observable
+      .take(1)
+      .filter(queryCache => queryCache == null)
+      .subscribe(() => queryBuilder.invoke());
+
```

Comment:
what is the purpose of subscribing to the observable above, why couldn't you just do queryBuilder.invoke()?

Reply:
I think this was meant to be something like `.subscribe(q => q.invoke())` maybe? (so that we can invoke any of the QueryBuilders from the request cache that haven't been fulfilled.  I may be misinterpreting this, though.